src/reader/mod.rs

```diff
@@ -21,42 +20,23 @@ mod parser;
 mod config;
 mod events;
 
+mod error_impl;
+
+#[derive(Debug)]
+pub enum ErrorKind {
```

Comment:
I thought about that but finally sticked to `error_impl` since `error::Something` seems to usually denote `std::error::Something`.

Currently there is no need to import `std::error` into the module, but if it is needed somewhen -- it may cause some minor problems.

What I am almost sure about is that `error_impl`'s content is not likely to be imported to anywhere.

Is there any idiomatic way to name modules created just to keep things separately?